client/src/main/scala/sbtcompendium/client/CompendiumClient.scala

```diff
@@ -112,6 +112,11 @@ object CompendiumClient {
               protocol <- retrieveProtocol(identifier, safeInt(v))
               code     <- handleAvro(protocol)
             } yield code
+          case IdlName.Protobuf =>
+            for {
+              protocol <- retrieveProtocol(identifier, safeInt(v))
+              code     <- handleProto(protocol)
+            } yield code
```

Comment:
Looks like you could do a single retrieveProtocolo:

```scala
retrieveProtocol(identifier, safeInt(v)).map {
  case IdlName.Avro => handleAvro(protocol)
  case IdlName.Protobuf => handleProto(protocol)
}
```

or something like that.

Reply:
Yes, you are right, but since not all schema types are implemented yet, I avoid making an http call before rising an error